lib/t/feeds.ex

```diff
@@ -73,29 +73,45 @@ defmodule T.Feeds do
       distance: distance
     } = feed_filter
 
-    feeded = FeededProfile |> where(for_user_id: ^user_id) |> select([s], s.user_id)
+    # feeded = FeededProfile |> where(for_user_id: ^user_id) |> select([s], s.user_id)
 
-    most_liked_count = count - div(count, 2)
+    # most_liked_count = count - div(count, 2)
 
-    most_liked =
-      most_liked_q(user_id, gender, gender_preferences, feeded)
-      |> maybe_apply_age_filters(min_age, max_age)
-      |> maybe_apply_distance_filter(location, distance)
-      |> limit(^most_liked_count)
-      |> Repo.all()
+    # most_liked_q(user_id, gender, gender_preferences, feeded)
+    # most_liked =
+    feed_profiles_q(user_id, gender, gender_preferences)
+    # |> maybe_apply_age_filters(min_age, max_age)
+    # |> maybe_apply_distance_filter(location, distance)
+    # |> limit(^most_liked_count)
+    # |> select_merge([p], %{distance: st_distance(^location, p.location) / 1000})
+    # select([p], %{p | distance: st_distance(^location, p.location) / 1000})
+    |> select([p], {p, fragment("st_distance(?, location) / 1000", ^location)})
+    |> order_by(fragment("st_distance(?, location) / 1000", ^location))
```

Comment:
So far I was not able to find a way to use normal order_by in Ecto, avoiding double computation (?)
related: https://stackoverflow.com/questions/58579900/order-by-virtual-computed-field

Reply:
note that st_distance is not in meters

Reply:
right, thanks!